py/qstr.c

```diff
@@ -171,14 +194,48 @@ STATIC qstr qstr_add(const byte *q_ptr) {
     return MP_STATE_VM(last_pool)->total_prev_len + MP_STATE_VM(last_pool)->len - 1;
 }
 
+#define MP_QSTR_SEARCH_THRESHOLD 10
```

Comment:
How was this number chosen?

Reply:
The rational was that sequential search over small sequences is more efficient than binary search because of caching, compiler optimizations etc. It's probably a different number for different architectures, compiler flags etc.  

So this number was chosen simply because it was good enough on my tests.